examples/common/pigweed/rpc_services/Device.h

```diff
@@ -270,14 +270,46 @@ class Device : public pw_rpc::nanopb::Device::Service<Device>
 
     virtual pw::Status SetPairingState(const chip_rpc_PairingState & request, pw_protobuf_Empty & response)
     {
+        chip::ChipError err;
         if (request.pairing_enabled)
         {
-            DeviceLayer::ConnectivityMgr().SetBLEAdvertisingEnabled(true);
-            DeviceLayer::ConnectivityMgr().SetBLEAdvertisingMode(DeviceLayer::ConnectivityMgr().kFastAdvertising);
+            if (!DeviceLayer::ConnectivityMgr().IsBLEAdvertisingEnabled())
+            {
+                err = DeviceLayer::ConnectivityMgr().SetBLEAdvertisingEnabled(true);
+                if (err != CHIP_NO_ERROR)
+                {
+                    ChipLogError(NotSpecified, "RPC SetPairingState failed to enable ble: %" CHIP_ERROR_FORMAT, err.Format());
+                    return pw::Status::Internal();
+                }
+            }
+            if (!chip::Server::GetInstance().GetCommissioningWindowManager().IsCommissioningWindowOpen())
+            {
+                DeviceLayer::StackLock lock;
+                err = chip::Server::GetInstance().GetCommissioningWindowManager().OpenBasicCommissioningWindow();
+                if (err != CHIP_NO_ERROR)
+                {
+                    ChipLogError(NotSpecified, "RPC SetPairingState failed to open commissioning window: %" CHIP_ERROR_FORMAT,
+                                 err.Format());
+                    return pw::Status::Internal();
+                }
+            }
         }
         else
         {
-            DeviceLayer::ConnectivityMgr().SetBLEAdvertisingEnabled(false);
+            if (DeviceLayer::ConnectivityMgr().IsBLEAdvertisingEnabled())
+            {
+                err = DeviceLayer::ConnectivityMgr().SetBLEAdvertisingEnabled(false);
```

Comment:
Again, CloseCommissioningWindow will do that....

Reply:
done